libs/shared/flagd-core/src/lib/feature-flag.ts

```diff
@@ -42,14 +91,82 @@ export class FeatureFlag {
     return this._defaultVariant;
   }
 
-  get targeting(): unknown {
-    return this._targeting;
-  }
-
   get variants(): Map<string, FlagValue> {
     return this._variants;
   }
 
+  get metadata(): FlagMetadata {
+    return this._metadata;
+  }
+
+  evaluate(evalCtx: EvaluationContext, logger: Logger = this.logger): RequiredResolutionDetails<JsonValue> {
+    let variant: string;
+    let reason: ResolutionReason;
+
+    if (this._targetingParseErrorMessage) {
+      return {
+        reason: StandardResolutionReasons.ERROR,
+        errorCode: ErrorCode.PARSE_ERROR,
+        errorMessage: this._targetingParseErrorMessage,
+        flagMetadata: this.metadata,
+      };
+    }
+
+    if (!this._targeting) {
+      variant = this._defaultVariant;
+      reason = StandardResolutionReasons.STATIC;
+    } else {
+      let targetingResolution: JsonValue;
+      try {
+        targetingResolution = this._targeting.evaluate(this._key, evalCtx, logger);
+      } catch (e) {
+        logger.debug(`Error evaluating targeting rule for flag '${this._key}': ${(e as Error).message}`);
+        return {
+          reason: StandardResolutionReasons.ERROR,
+          errorCode: ErrorCode.GENERAL,
+          errorMessage: `Error evaluating targeting rule for flag '${this._key}'`,
+          flagMetadata: this.metadata,
+        };
+      }
+
+      // Return default variant if targeting resolution is null or undefined
+      if (targetingResolution === null || targetingResolution === undefined) {
+        variant = this._defaultVariant;
+        reason = StandardResolutionReasons.DEFAULT;
+      } else {
+        // Obtain resolution in string. This is useful for short-circuiting json logic
+        variant = targetingResolution.toString();
+        reason = StandardResolutionReasons.TARGETING_MATCH;
+      }
+    }
+
+    // if (typeof variant !== 'string') {
```

Comment:
I don't think this check is necessary. I wasn't able to write a test that hit this code.